week3/golf.hs

```diff
@@ -0,0 +1,33 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module Golf where
+
+import Data.List
+  
+-- Exercise 1
+skips :: [a] -> [[a]]
+skips [] = []
+skips a@(_:xs) = a : skips xs
+
+-- Exercise 2
+bundle :: Int -> [Integer] -> [[Integer]]
+bundle _ [] = []
+bundle n a@(_:xs) = filter (\x -> length x == n) 
+                    ((take n a) : bundle n xs)
+
+localMaxima :: [Integer] -> [Integer]
+localMaxima = map (\(_:b:_) -> b) 
+              . filter (\(a:b:c:_) -> (b > a) && (b > c)) 
+              . bundle 3
```

Comment:
This is just a suggestion if you want to try and make this into a single function:
you are using `bundle` that takes `n` as a variable, but `n` never actually changes, it's just `3`, this is a hint that you might be able to encode the same logic using pattern-matching on the input instead.

For instance `localMaxima (x:xs@(y:z:_)) = ...` matches only lists with at least 3 elements. But you might have to use recursion to get this to work without an accumulator.

Reply:
Refactored with pattern match